### DISCUSSION TITLE
Read absolute finger positions on touchpad

### DISCUSSION THREAD

**Discussion by realmayus** [rust-windowing/winit#3706] [Q&A] [answered]
Is it possible to read the absolute position of fingers within a touchpad's "coordinate space"?

I want to create something similar to Apple's Preview app's signature writing feature: when you sign, you do that within the entire rectangle of your touchpad, which is much easier than dragging the cursor like when you're using a mouse.

**Comment by madsmtm** [comment, marked as answer]
I think you have to somehow get the `NSEvent` of each touch, and from there the the `allTouches` array of `NSTouch`es, and then use `normalizedPosition` on that.

Not something you can do with Winit, I'd suggest you create and manage a `NSView` yourself, and then attach it as a subview of the Winit view.
